treeherder/etl/management/commands/ingest_push_and_tasks.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import logging
+
+import aiohttp
+import taskcluster
+import taskcluster.aio
+from django.core.management.base import BaseCommand
+
+from treeherder.etl.job_loader import JobLoader
+from treeherder.etl.taskcluster_pulse.handler import (EXCHANGE_EVENT_MAP,
+                                                      handleMessage)
+
+logger = logging.getLogger(__name__)
+rootUrl = "https://taskcluster.net"
+options = {"rootUrl": rootUrl}
+loop = asyncio.get_event_loop()
+# Limiting the connection pool just in case we have too many
+conn = aiohttp.TCPConnector(limit=60)
+# Remove default timeout limit of 5 minutes
+timeout = aiohttp.ClientTimeout(total=0)
+session = taskcluster.aio.createSession(loop=loop, connector=conn, timeout=timeout)
+asyncQueue = taskcluster.aio.Queue(options, session=session)
+
+stateToExchange = {}
+for key, value in EXCHANGE_EVENT_MAP.items():
+    stateToExchange[value] = key
+
+
+async def handleTaskId(taskId):
+    results = await asyncio.gather(asyncQueue.status(taskId), asyncQueue.task(taskId))
+    await handleTask({
+        "status": results[0]["status"],
+        "task": results[1],
+    })
+
+
+async def handleTask(task):
+    taskId = task["status"]["taskId"]
+    runs = task["status"]["runs"]
+    for run in runs:
+        message = {
+            "exchange": stateToExchange[run["state"]],
+            "payload": {
+                "status": {
+                    "taskId": taskId,
+                    "runs": runs,
+                },
+                "runId": run["runId"],
+            }
+        }
+        try:
+            tc_th_message = await handleMessage(message, task["task"])
+            if tc_th_message:
+                logger.info("Loading into DB:\t%s/%s", taskId, run["runId"])
+                # XXX: This seems our current bottleneck
+                JobLoader().process_job(tc_th_message)
```

Comment:
When I run this script `docker-compose run backend bash -c "./manage.py ingest_push_and_tasks"` I see hundreds of messages that happen within `handleMessage` and later on I will start seeing messages from `JobLoader().process_job(tc_th_message)`.

I wish that all `process_job` were not run last since they become sequential inserts into the DB and are not running concurrently.

Any suggestions on what to try?

Reply:
I think this is bearable. 6 minutes to ingest all tasks for a push is not the end of the world.